src/main/java/net/snowflake/client/jdbc/ArrowBatch.java

```diff
@@ -0,0 +1,10 @@
+package net.snowflake.client.jdbc;
+
+import java.util.List;
+import org.apache.arrow.vector.VectorSchemaRoot;
+
+public interface ArrowBatch {
+  List<VectorSchemaRoot> fetch() throws SnowflakeSQLException;
+
+  long getRowCount();
```

Comment:
I'm wondering whether we should use general word 'row' or we might use 'getVectorCount'?

Reply:
Same as above: we care about rows, not vectors (which are columns). Maybe we'd like to know the number of columns in the batch, but it should be a separate function if so. Note: the column count can be easily taken from individual `VectorSchemaRoots` returned by fetch()

Reply:
Ok.